internal/slicer/slicer.go

```diff
@@ -158,13 +158,31 @@ func Run(options *RunOptions) (*Report, error) {
 			Package:   slice.Package,
 			Extract:   extract[slice.Package],
 			TargetDir: targetDir,
-			Globbed:   globbedPaths,
 			// Creates the filesystem entry and adds it to the report.
-			Create: func(o *fsutil.CreateOptions) error {
+			Create: func(extractInfo *deb.ExtractInfo, o *fsutil.CreateOptions) error {
 				entry, err := fsutil.Create(o)
 				if err != nil {
 					return err
 				}
+
+				// We only want to keep the entries that were explicitly listed
+				// in the slice definition.
+				if extractInfo == nil {
+					return nil
+				}
+				if _, ok := slice.Contents[extractInfo.Path]; !ok {
```

Comment:
How does that happen?

Reply:
It happens in several places. The first case is when we have a text file, to preserve parent directories we add a optional ExtractInfo of its parent directory to be passed to the extractor ([pointer](https://github.com/canonical/chisel/blob/main/internal/slicer/slicer.go#L121-L124)). The second case is copyright files that are also added as optional and passed to the extractor ([pointer](https://github.com/canonical/chisel/blob/main/internal/slicer/slicer.go#L128-L131)). In both cases we dont want to include the file in the report because it is not listed explicitly in the slice definition.

I have thought about dropping these lines and relying on how we are going to handle parent directories and copyright slices, but I think it would be a mistake. In my opinion it is better to have the extra flexibility and not depend on two different things which are not in the code right now and whose requirements can change in the future. It seems right to be able to pass more things to be extracted and those not being included in the report.

Reply:
Discussed in the meeting, marking it as solved.